cmd2/cmd2.py

```diff
@@ -4230,13 +4230,13 @@ def do_history(self, args: argparse.Namespace) -> Optional[bool]:
                 self.perror("Cowardly refusing to run all previously entered commands.")
                 self.perror("If this is what you want to do, specify '1:' as the range of history.")
             else:
-                return self.runcmds_plus_hooks(history)
+                return self.runcmds_plus_hooks(list(history.values()))
```

Comment:
Something just smells off to me.  The `History` class is still just a `List[HistoryItem]` with some added methods.  But the `cmd2.Cmd._get_history()` method returns a `Dict[int, HistoryItem]` and does so using the `self.history` History list.  

It feels awkward and like we don't have a data structure or algorithm right somewhere.

Should the `History` class be modified to be a dictionary?  Or should we add a convenience method to it so we don't need this method that returns an intermediate dicionary and then re-converts that back to a list?

Reply:
Ok, I've slept on this and don't have any brilliant ideas.

Perhaps if we just encapsulated more of the weird functionality within the `History` class it would feel less awkward?  Not sure.  What do you think?

Reply:
Yeah, I couldn't think of a better structure either. We need the results and we need to know what their indexes are. Since the user is searching all of the history, results won't be necessarily be contiguous from the main list. At least the dictionary makes it easy to treat the values as a list. And thankfully the dictionaries are ordered by index numbers.